File: client/bcephal-client-model/Conditions/ConditionalExpressionItem.cs
using Bcephal.Models.Base;
using Bcephal.Models.Filters;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Conditions
{
    public class ConditionalExpressionItem : Persistent
    {
        public string FilterVerb { get; set; }

        public string OpenBrackets { get; set; }

        public string CloseBrackets { get; set; }

        public string Operator { get; set; }

        public decimal? Value { get; set; }

        public long? SpotId1 { get; set; }

        public long? SpotId2 { get; set; }

        public DimensionType Value2Type { get; set; }

        public int Position { get; set; }

        public override int CompareTo(object obj)
        {
            if (obj == null || !(obj is ConditionalExpressionItem)) return 1;
            if (this == obj) return 0;
            if (this == obj || !this.Id.HasValue) return 0;
            if (this.Id.HasValue && this.Id.Equals(((ConditionalExpressionItem)obj).Id)) return 0;
            return this.Position.CompareTo(((ConditionalExpressionItem)obj).Position);
        }

        public ConditionalExpressionItem Copy()
        {
            ConditionalExpressionItem copy = new ConditionalExpressionItem();
            copy.Operator = this.Operator;
            copy.Value = this.Value;
            copy.Id = this.Id;
            copy.SpotId1 = this.SpotId1;
            copy.SpotId2 = this.SpotId2;
            copy.Value2Type = this.Value2Type;
            copy.FilterVerb = this.FilterVerb;
            copy.OpenBrackets = this.OpenBrackets;
            copy.CloseBrackets = this.CloseBrackets;
            copy.Position = this.Position;
            return copy;
        }
    }
}

File: server/bcephal-base/src/main/java/com/moriset/bcephal/domain/condition/ConditionalExpressionItem.java
/**
 * 
 */
package com.moriset.bcephal.domain.condition;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.Persistent;
import com.moriset.bcephal.domain.dimension.DimensionType;
import com.moriset.bcephal.domain.filters.FilterVerb;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString.Exclude;

/**
 * @author Joseph Wambo
 *
 */
@Entity(name = "ConditionalExpressionItem")
@Table(name = "BCP_CONDITIONAL_EXPRESSION_ITEM")
@Data
@EqualsAndHashCode(callSuper = false)
public class ConditionalExpressionItem extends Persistent {

	private static final long serialVersionUID = -486996300179306333L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "conditional_expression_item_seq")
	@SequenceGenerator(name = "conditional_expression_item_seq", sequenceName = "conditional_expression_item_seq", initialValue = 1, allocationSize = 1)
	private Long id;
	
	@Exclude
	@JsonIgnore
	@ManyToOne
	@JoinColumn(name = "expression")
	private ConditionalExpression expression;

	@Enumerated(EnumType.STRING)
	private FilterVerb filterVerb;

	private String openBrackets;

	private String closeBrackets;

	private int position;

	private String operator;
	
	private BigDecimal value;
	
	private Long spotId1;
	
	private Long spotId2;
	
	@Enumerated(EnumType.STRING)
	private DimensionType value2Type;

	
	public ConditionalExpressionItem() {
		this.filterVerb = FilterVerb.AND;
	}
	
	
	@Override
	public Persistent copy() {
		return null;
	}
	
}
